File: node/examples/stream-firehose.ts
import { RtprWebSocket } from "@rtpr-io/rtpr";

const ws = new RtprWebSocket("your_api_key");

ws.onArticle((article) => {
  const time = new Date().toLocaleTimeString();
  console.log(`[${time}] ${article.ticker}: ${article.title}`);
  console.log(`  Source: ${article.author}`);
  console.log();
});

ws.onConnect(() => {
  console.log("Connected to RTPR WebSocket");
});

ws.onDisconnect((data) => {
  console.log("Disconnected:", data);
});

ws.onError((data) => {
  console.error("Error:", data);
});

// Subscribe to all tickers (firehose mode)
// Use ["AAPL", "TSLA"] to subscribe to specific tickers
ws.connect(["*"]).catch(console.error);

File: python/examples/stream_firehose.py
"""Stream all press releases in real time via WebSocket."""

import asyncio

from rtpr import RtprWebSocket

API_KEY = "your_api_key"

ws = RtprWebSocket(API_KEY)


@ws.on_article
async def handle_article(article):
    print(f"[{article.ticker}] {article.title}")
    print(f"  Source: {article.author}")
    print(f"  Body:   {article.article_body[:120]}...")
    print()


@ws.on_connect
async def handle_connect(data):
    print("Connected to RTPR real-time feed")


asyncio.run(ws.connect(tickers=["*"]))
